consensus/parlia/parlia.go

```diff
@@ -1840,15 +1843,20 @@ func (p *Parlia) distributeIncoming(val common.Address, state *state.StateDB, he
 	if balance.Cmp(common.U2560) <= 0 {
 		return nil
 	}
-	state.SetBalance(consensus.SystemAddress, common.U2560, tracing.BalanceChangeUnspecified)
-	state.AddBalance(coinbase, balance, tracing.BalanceChangeUnspecified)
+
+	// FIXME Those need some balance change reason, in our current Firehose implementation we had put
+	// firehose.BalanceChangeReason("reward_transfaction_fee"), this would map to
+	// `tracing.BalanceIncreaseRewardTransactionFee` in the new tracing API, is that accurate enough here?
+	// I would be fine adding a new more specific reason if needed.
+	state.SetBalance(consensus.SystemAddress, common.U2560, tracing.BalanceIncreaseRewardTransactionFee)
+	state.AddBalance(coinbase, balance, tracing.BalanceIncreaseRewardTransactionFee)
```

Comment:
@maoueh I think `BalanceIncreaseRewardTransactionFee` is not accurate. I would prefer to introduce the two new reason for BSC, but I am afraid of forward compatible issue, how about:
```
BalanceDecreaseBSCDistributeReward BalanceChangeReason = 1000
BalanceIncreaseBSCDistributeReward BalanceChangeReason = 1001
```

Reply:
I prefer indeed to add 2 new reasons, makes tracing system aware of important balance changes. Don't think Unspecified is good here.

> but I am afraid of forward compatible issue

Indeed, that's always a possibility, even more that the `BalanceChangeReason` is a `byte` today leading to just 255 values.

But having managed multiple Geth forks over the last 3 years, I can tell you that new balance change reasons is very rare. I think it will be just fine. I've put BSC starting at 210 for now, let me know how it feels.